Tasks/KubernetesManifestV0/Tests/L0.ts

```diff
@@ -275,6 +275,11 @@ describe('Kubernetes Manifests Suite', function () {
 
         assert(bigSecondYaml.spec.template.spec.containers[0].image === 'nginx:42', 'nginx image not tagged correctly');
         assert(bigSecondYaml.spec.template.spec.initContainers[0].image === 'nginx-init:42.1', 'nginx-init image not tagged correctly');
+        
+        const untaggedImages = utils.substituteImageNameInSpecFile(deploymentFile, 'mysql', 'mysql:8.0');
+        const untaggedImagesYaml = yaml.load(smallNameEditSecond);
+        assert(untaggedImagesYaml.spec.template.spec.containers[0].image === 'mysql:8.0', 'untagged image not tagged correctly');
```

Comment:
This is the correct test implementation, and it passes.
```ts
        const untaggedImages = utils.substituteImageNameInSpecFile(deploymentFile, 'mysql', 'mysql:8.0');
        const untaggedImagesYaml = yaml.load(untaggedImages);
        assert(untaggedImagesYaml.spec.template.spec.containers[2].image === 'mysql:8.0', 'untagged image not tagged correctly');
```

However, I came here because I also have lost tags on my deployments. It stopped working on July 17, without changes in the manifest or pipeline.

The images are untagged in the manifest, and this task does not add tags:
````
yaml
  - task: KubernetesManifest@0
    inputs:
      action: 'deploy'
      kubernetesServiceConnection: 'con1'
      namespace: 'preproduction'
      manifests: '$(System.ArtifactsDirectory)/.kubernetes/preprod.yml'
      containers: |
        example.azurecr.io/nginx:$(Build.BuildNumber)
        example.azurecr.io/server:$(Build.BuildNumber)
```

Reply:
Thanks for the hint. I updated the test and added another one for images with registries.
Tests are passing, but I’m still getting the same behavior as you are.